app/earn/page.tsx

```diff
@@ -16,13 +18,22 @@ export default async function Page() {
     redirect('/')
 
   const vaultsByChain = await fetchVaults()
+  const pastYearPrices = await fetchPastYearPrices()
   return (
     <div className={styles.page}>
-      <VaultGrid vaultsByChain={vaultsByChain} />
+      <VaultGrid vaultsByChain={vaultsByChain} pastYearPrices={pastYearPrices} />
     </div>
   )
 }
 
+async function fetchPastYearPrices(): Promise<PastYearPrices> {
+  const symbols = Object.values(TokenOrder)
+  const result = {} as PastYearPrices
+  for (const symbol of symbols)
+    result[symbol] = await getPastYearPrice(symbol)
```

Comment:
Maybe it mus be run in parallel? Or if it done to decrease amount of requests than maybe need soome delay/retry logic?